gsy_e_sdk/clients/rest_asset_client.py

```diff
@@ -0,0 +1,171 @@
+import logging
+from concurrent.futures.thread import ThreadPoolExecutor
+from typing import Dict
+
+from gsy_framework.client_connections.utils import (
+    RestCommunicationMixin, blocking_post_request, log_market_progression,
+    retrieve_jwt_key_from_server)
+from gsy_framework.client_connections.websocket_connection import WebsocketThread
+from gsy_framework.utils import execute_function_util
+
+from gsy_e_sdk import APIClientInterface
+from gsy_e_sdk.constants import MAX_WORKER_THREADS
+from gsy_e_sdk.utils import (
+    domain_name_from_env, get_aggregator_prefix, get_configuration_prefix, log_trade_info,
+    logging_decorator, simulation_id_from_env, websocket_domain_name_from_env)
+from gsy_e_sdk.websocket_device import DeviceWebsocketMessageReceiver
+
+
+REGISTER_COMMAND_TIMEOUT = 15 * 60
+
+#pylint: disable=too-many-instance-attributes
+class RestAssetClient(APIClientInterface, RestCommunicationMixin):
+    """Client class for assets to be used while working with REST."""
+
+    # pylint: disable=super-init-not-called
+    # pylint: disable=too-many-arguments
+    def __init__(self, area_id, simulation_id=None, domain_name=None, websockets_domain_name=None,
```

Comment:
I think it is a good time time rename this variable too:
```suggestion
    def __init__(self, asset_uuid, simulation_id=None, domain_name=None, websockets_domain_name=None,
```

Reply:
This would break backwards compatibility for users that used the old parameter, like:
```python
RestDeviceClient(area_id=123, ...)
```
One way to overcome this would be to explicitly init the new `RestAssetClient` in the old `RestDeviceClient` init method. For example:
```python
class RestDeviceClient(RestAssetClient):
    def __init__(
            self, self, area_id, simulation_id=None, domain_name=None, websockets_domain_name=None,
            autoregister=False, start_websocket=True, sim_api_domain_name=None):
        super().__init__(
            area_id=asset_uuid, simulation_id=simulation_id, domain_name=domain_name, 
            websockets_domain_name=websockets_domain_name,
            autoregister=False, start_websocket=True, sim_api_domain_name=sim_api_domain_name)
```
This would make the inheritance process more complex to maintain, because we would need to make sure that every time the parameters are changed in `RestAssetClient` we also change them in the old `RestDeviceClient`.

If you want I can do that, I just want to make sure that this is what we want.

Reply:
Ah yes, backwards compatibility, I forgot. Having in mind, that we deprecate the old clients at some point anyway I think it is better to present a consistent interface to the user and bite the bullet to have more maintenance to do if we change parameters here (which I do not predict).
TLDR; IMO your proposed change LGTM with one little typo found and corrected here:
```python
class RestDeviceClient(RestAssetClient):
    def __init__(
            self, self, area_id, simulation_id=None, domain_name=None, websockets_domain_name=None,
            autoregister=False, start_websocket=True, sim_api_domain_name=None):
        super().__init__(
            asset_uuid=area_id, simulation_id=simulation_id, domain_name=domain_name, 
            websockets_domain_name=websockets_domain_name,
            autoregister=False, start_websocket=True, sim_api_domain_name=sim_api_domain_name)
```

Reply:
Thanks, I've added the changes in e33a06a 👍 

BTW: the same could be applied to the `RedisDeviceClient` class. However, that class inherits from `RedisClientBase` which is not deprecated. So changing the parameters there would introduce some incoherent changes in my opinion (or we would need to change `RedisClientBase` to support this in a backward-compatible manner). I left it out for the moment, let me know if I should work on it.